From b0062b0a35ecc27da6ab2d655391438788f42ee1 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Thu, 15 May 2014 09:11:52 +0200
Subject: [LIBREPORT PATCH] ureport: add support for client-side authentication

Please note that the libreport_curl api is changed and since we're not
bumping sonames ABRT has to explicitly depend on this version in spec.

Related to rhbz#1053042.

Signed-off-by: Martin Milata <mmilata@redhat.com>

Conflicts:

	src/lib/json.c
	src/plugins/ureport.c
---
 src/lib/abrt_curl.c | 7 +++++++
 src/lib/json.c      | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/src/lib/abrt_curl.c b/src/lib/abrt_curl.c
index 35a7546..5b1c562 100644
--- a/src/lib/abrt_curl.c
+++ b/src/lib/abrt_curl.c
@@ -499,6 +499,13 @@ abrt_post(abrt_post_state_t *state,
         xcurl_easy_setopt_long(handle, CURLOPT_SSL_VERIFYPEER, 0);
         xcurl_easy_setopt_long(handle, CURLOPT_SSL_VERIFYHOST, 0);
     }
+    if (state->client_cert_path && state->client_key_path)
+    {
+        xcurl_easy_setopt_ptr(handle, CURLOPT_SSLCERTTYPE, "PEM");
+        xcurl_easy_setopt_ptr(handle, CURLOPT_SSLKEYTYPE, "PEM");
+        xcurl_easy_setopt_ptr(handle, CURLOPT_SSLCERT, state->client_cert_path);
+        xcurl_easy_setopt_ptr(handle, CURLOPT_SSLKEY, state->client_key_path);
+    }
 
     // This is the place where everything happens.
     // Here errors are not limited to "out of memory", can't just die.
diff --git a/src/lib/json.c b/src/lib/json.c
index 314085b..d0830cd 100644
--- a/src/lib/json.c
+++ b/src/lib/json.c
@@ -84,6 +84,12 @@ struct abrt_post_state *post_ureport(const char *json, struct ureport_server_con
         post_state->client_key_path = config->ur_client_key;
     }
 
+    if (config->ur_client_cert && config->ur_client_key)
+    {
+        post_state->client_cert_path = config->ur_client_cert;
+        post_state->client_key_path = config->ur_client_key;
+    }
+
     static const char *headers[] = {
         "Accept: application/json",
         "Connection: close",
-- 
1.8.3.1

